qiskit_nature/problems/second_quantization/electronic/electronic_structure_problem.py

```diff
@@ -70,25 +68,16 @@ def second_q_ops(self) -> List[SecondQuantizedOp]:
             operator, and (if available) x, y, z dipole operators.
         """
         self._molecule_data = cast(QMolecule, self.driver.run())
-        self._molecule_data_transformed = cast(QMolecule, self._transform(self._molecule_data))
-
-        # TODO: in a follow-up PR we should gather these properties in a super-object. Possibly
-        # ElectronicDriverResult?
-        properties = []
-        for cls in [
-            ElectronicEnergy,
-            ParticleNumber,
-            AngularMomentum,
-            Magnetization,
-            TotalDipoleMoment,
-        ]:
-            prop = cls.from_driver_result(self._molecule_data_transformed)  # type: ignore
-            if prop is not None:
-                properties.append(prop)
-
-        second_quantized_ops_list = reduce(
-            lambda a, b: a + b, [prop.second_q_ops() for prop in properties]
-        )
+        if self._legacy_transform:
+            qmol_transformed = self._transform(self._molecule_data)
+            self._driver_result_transformed = ElectronicDriverResult.from_legacy_driver_result(
```

Comment:
There are a two aspects here:

#### configurability of which properties to compute
Yes, this is yet to come about, because currently the `ElectronicDriverResult` hard-codes the list of properties inside of `second_q_ops` (there is an open TODO there). However, a user can also construct a `GroupedProperty` themselves. I will actually adapt the transformer interface slightly to work with `GroupedProperty` in general and not require a `DriverResult` as this will make it easier to be used by someone who does not want to rely on a driver.

#### ParticleNumber
In the specific case of the `ActiveSpaceTransformer` (and `FreezeCore` by extension) the `ParticleNumber` property is a hard requirement! I will make that more clear in the documentation and the code.

Reply:
Ok, since as I mentioned I still consider these properties becoming operators that are measured, where if they end up as aux-ops the measurement is optional - I guess the presence of the properties and whether we want to measure them or not, is kinda muddied in my mind. I had thought the properties would not be created if were are not interested in measuring them. But the particle number has this duality as we discussed before, whereby we have input data that has the value and we need that to configure things elsewhere. Since there is more to do in this regard, as you mention, lets see how things progress in the next steps then